wrangler-service/src/main/java/co/cask/wrangler/service/common/AbstractWranglerService.java

```diff
@@ -71,4 +120,98 @@ protected boolean shouldCopyHeader(@Nullable String workspaceId) throws Workspac
       .map(String::trim)
       .anyMatch(directive -> directive.startsWith("parse-as-csv") && directive.endsWith("true"));
   }
+  
+  public UserGroupInformation getProxyUGI(String impersonatedUser) throws IOException, 
+                           InterruptedException, ExecutionException {
+    UserGroupInformation defaultUgi = null;
+    if (!UserGroupInformation.isSecurityEnabled()) {
+      LOG.debug("Security is disabled");
+      return defaultUgi;
+    }
+        
+    boolean userImpersonationEnabled = isUserImpersonationEnabled();
+    if (!userImpersonationEnabled || (impersonatedUser == null)) {
```

Comment:
if userImpersonationEnabled, impersonatedUser should be mandatory

Reply:
Done modifications. Thanks